src/main/java/pl/edu/uj/notes/user/UserService.java

```
@@ -20,4 +20,12 @@ public int createUser(CreateUserRequest request) {
     userRepository.save(user);
     return user.getId();
   }
+
+  public void deleteUser(int id) {
+    UserEntity user =
+        userRepository
+            .findById(id)
+            .orElseThrow(() -> new IllegalArgumentException("User not found"));
```

Let's create dedicated exception for this. Should we throw 404 when user is not found?

---

Created UserNotFoundException. It now throws 404 Not Found using @ResponseStatus(HttpStatus.NOT_FOUND).